src/libraries/System.Private.CoreLib/src/System/Numerics/Decimal128.cs

```diff
@@ -0,0 +1,198 @@
+        {
+            NumberFormatInfo.ValidateParseStyleFloatingPoint(style);
+            return Number.ParseDecimal128(s, style, NumberFormatInfo.GetInstance(provider));
+        }
+        public static Decimal128 Parse(string s, NumberStyles style, IFormatProvider? provider)
+        {
+            if (s is null)
+            {
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.s);
+            }
+            return Parse(s.AsSpan(), style, provider);
+        }
+
+        public static bool TryParse([NotNullWhen(true)] string? s, out Decimal128 result) => TryParse(s, NumberStyles.Number, provider: null, out result);
+        public static bool TryParse(ReadOnlySpan<char> s, out Decimal128 result) => TryParse(s, NumberStyles.Number, provider: null, out result);
+        public static bool TryParse(ReadOnlySpan<char> s, IFormatProvider? provider, [MaybeNullWhen(false)] out Decimal128 result) => TryParse(s, NumberStyles.Number, provider, out result);
+        public static bool TryParse([NotNullWhen(true)] string? s, IFormatProvider? provider, [MaybeNullWhen(false)] out Decimal128 result) => TryParse(s, NumberStyles.Number, provider, out result);
+        public static bool TryParse(ReadOnlySpan<char> s, NumberStyles style, IFormatProvider? provider, [MaybeNullWhen(false)] out Decimal128 result)
+        {
+            NumberFormatInfo.ValidateParseStyleFloatingPoint(style);
+            return Number.TryParseDecimal128(s, style, NumberFormatInfo.GetInstance(provider), out result) == Number.ParsingStatus.OK;
+        }
+        public static bool TryParse([NotNullWhen(true)] string? s, NumberStyles style, IFormatProvider? provider, [MaybeNullWhen(false)] out Decimal128 result)
+        {
+            NumberFormatInfo.ValidateParseStyleFloatingPoint(style);
+
+            if (s == null)
+            {
+                result = new Decimal128(0, 0);
+                return false;
+            }
+            return Number.TryParseDecimal128(s.AsSpan(), style, NumberFormatInfo.GetInstance(provider), out result) == Number.ParsingStatus.OK;
+        }
+
+        public int CompareTo(object? value)
+        {
+            if (value == null)
+            {
+                return 1;
+            }
+
+            if (value is not Decimal128 i)
+            {
+                throw new ArgumentException(SR.Arg_MustBeDecimal128);
+            }
+
+            return Number.CompareDecimalIeee754<Decimal128, Int128, UInt128>(_value, i._value);
+        }
+
+        public bool Equals(Decimal128 other)
+        {
+            return Number.CompareDecimalIeee754<Decimal128, Int128, UInt128>(_value, other._value) == 0;
+        }
+
+        public override bool Equals([NotNullWhen(true)] object? obj)
+        {
+            return obj is Decimal128 && Equals((Decimal128)obj);
+        }
+
+        public override int GetHashCode()
+        {
+            return _value.GetHashCode();
+        }
```

Comment:
+0 and -0 should have the same hash code. The easiest way can be stripping the sign bit.

I also remember that there can be same values with different representations. https://github.com/dotnet/runtime/blob/9b57a265c7efd3732b035bade005561a04767128/src/libraries/System.Private.CoreLib/src/System/Numerics/TotalOrderIeee754Comparer.cs#L125-L132

Reply:
@huoyaoyuan  Thanks for your review. However this PR targets only the first phase as following: https://github.com/dotnet/runtime/issues/81376#issuecomment-1863903078 so I haven't added `NegativeZero` and `PositiveZero`. I will update `GetHashCode` method in the next phase.